CMakeLists.txt

```diff
@@ -1,26 +1,9 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
-project(prime_server LANGUAGES CXX C VERSION 0.6.7)
+include(cmake/version.cmake)
+get_version(${CMAKE_SOURCE_DIR}/prime_server/prime_server.hpp "PRIME_SERVER_")
+project(prime_server LANGUAGES CXX C VERSION ${PRIME_SERVER_VERSION_MAJOR}.${PRIME_SERVER_VERSION_MINOR}.${PRIME_SERVER_VERSION_PATCH})
```

Comment:
the prefix above `PRIME_SERVER_` becomes part of the variable name here. good news is, if somethings messed up cmake fails on invalid version string so that is nice!